src/compiler/types.ts

```diff
@@ -3031,7 +3031,7 @@ namespace ts {
         // Options
         NoTruncation                            = 1 << 0,   // Don't truncate result
         WriteArrayAsGenericType                 = 1 << 1,   // Write Array<T> instead T[]
-        // empty space
+        GenerateNamesForShadowedTypeParams      = 1 << 2,   // When a type parameter T is shadowing another T, generate a name for it so it can still be referenced

```

Comment:
Why does this need a flag?

Reply:
We don't wanna bother renaming the conflicts in quickinfo and signature help, at least in my opinion.